packages/foam-vscode/src/features/backlinks.ts

```diff
@@ -0,0 +1,174 @@
+import * as vscode from 'vscode';
+import { groupBy } from 'lodash';
+import {
+  Foam,
+  FoamWorkspace,
+  IDataStore,
+  isNote,
+  NoteLink,
+  Resource,
+  isSameUri,
+  URI,
+} from 'foam-core';
+import { getNoteTooltip } from '../utils';
+import { FoamFeature } from '../types';
+import { ResourceTreeItem } from '../utils/grouped-resources-tree-data-provider';
+import { Position } from 'unist';
+
+const feature: FoamFeature = {
+  activate: async (
+    context: vscode.ExtensionContext,
+    foamPromise: Promise<Foam>
+  ) => {
+    const foam = await foamPromise;
+
+    const workspacesURIs = vscode.workspace.workspaceFolders.map(
+      dir => dir.uri
+    );
+
+    const provider = new BacklinksTreeDataProvider(
+      foam.workspace,
+      foam.services.dataStore
+    );
+
+    vscode.window.onDidChangeActiveTextEditor(async () => {
+      provider.target = vscode.window.activeTextEditor?.document.uri;
+      await provider.refresh();
+    });
+
+    context.subscriptions.push(
+      vscode.window.registerTreeDataProvider('foam-vscode.backlinks', provider),
+      foam.workspace.onDidAdd(provider.refresh),
+      foam.workspace.onDidUpdate(provider.refresh),
+      foam.workspace.onDidDelete(provider.refresh)
+    );
+  },
+};
+export default feature;
+
+const isBefore = (a: Position, b: Position) =>
+  a.start.column - b.start.column || a.start.line - b.start.line;
+
+export class BacklinksTreeDataProvider
+  implements vscode.TreeDataProvider<BacklinkPanelTreeItem> {
+  public target?: URI = undefined;
```

Comment:
the reason we use `target` instead of checking the active editor inside here is that it makes it more easily testable, even though that means that we can't just refresh on active editor change, but also assign it to `target` (see line 35)